powerline/bindings/bar/powerline-bar.py

```diff
@@ -37,16 +39,20 @@ def render(event=None, data=None, sub=None):
 	args = parser.parse_args()
 	powerline = BarPowerline()
 
-	interval = 0.5
 	lock = Lock()
 
 	write = get_unicode_writer(encoding='utf-8')
+	render(reschedule=True)
 
 	if args.i3:
-		import i3
-		sub = i3.Subscription(render, 'workspace')
-
-	while True:
-		start_time = monotonic()
-		render()
-		time.sleep(max(interval - (monotonic() - start_time), 0.1))
+		try:
+			import i3ipc
+			conn = i3ipc.Connection()
```

Comment:
you mean like

    try: import i3ipc
    except ImportError: # use i3
    else: # use i3ipc

?